Silo doesn't link on Bridges2

Example below:
```
[I]bryngel@r007 example_cases/1D_bubblescreen $ python2 input.py post_process
dt:  0.013648856934

post_process>> Preparing serial job ...


post_process>> Serial job in progress ...

../../src/post_process_code/post_process: error while loading shared libraries: libsiloh5.so: cannot open shared object file: No such file or directory
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
../../src/post_process_code/post_process: error while loading shared libraries: libsiloh5.so: cannot open shared object file: No such file or directory
post_process>> Serial job completed!
```

Possibly related. When one runs `make check` in the SILO `src` directory I get an error: 

```
[I]bryngel@r007 src/SILO $ make check
Making check in .
make[1]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO'
make[1]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO'
Making check in src
make[1]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src'
make  check-recursive
make[2]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src'
Making check in score
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/score'
make[3]: Nothing to be done for 'check'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/score'
Making check in pdb
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/pdb'
make  check-am
make[4]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/pdb'
make[4]: Nothing to be done for 'check-am'.
make[4]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/pdb'
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/pdb'
Making check in silo
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/silo'
make  check-am
make[4]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/silo'
make[4]: Nothing to be done for 'check-am'.
make[4]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/silo'
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/silo'
Making check in debug
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/debug'
make[3]: Nothing to be done for 'check'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/debug'
Making check in netcdf
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/netcdf'
make[3]: Nothing to be done for 'check'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/netcdf'
Making check in pdb_drv
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/pdb_drv'
make[3]: Nothing to be done for 'check'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/pdb_drv'
Making check in hdf5_drv
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/hdf5_drv'
make[3]: Nothing to be done for 'check'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/hdf5_drv'
Making check in taurus
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/taurus'
make[3]: Nothing to be done for 'check'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/taurus'
Making check in zfp-0.5.5
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/zfp-0.5.5'
make[3]: Nothing to be done for 'check'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/zfp-0.5.5'
Making check in unknown
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/unknown'
make[3]: Nothing to be done for 'check'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/unknown'
Making check in filters
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/filters'
make[3]: Nothing to be done for 'check'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src/filters'
Making check in .
make[3]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src'
make[3]: Nothing to be done for 'check-am'.
make[3]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src'
make[2]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src'
make[1]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/src'
Making check in tools
make[1]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools'
Making check in .
make[2]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools'
make[2]: Nothing to be done for 'check-am'.
make[2]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools'
Making check in silock
make[2]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools/silock'
make[2]: Nothing to be done for 'check'.
make[2]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools/silock'
Making check in browser
make[2]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools/browser'
make[2]: Nothing to be done for 'check'.
make[2]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools/browser'
Making check in python
make[2]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools/python'
make[2]: Nothing to be done for 'check'.
make[2]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools/python'
make[1]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tools'
Making check in tests
make[1]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tests'

NOTICE: Silo test cases are not compiled by default.  To compile and test
NOTICE: them run "make check" from the silo "tests" directory. Otherwise
NOTICE: you can compile them individually with "make TESTNAME".  The test
NOTICE: names are: array, dir, extface, multi_test, partial_io, point, quad,
NOTICE: simple, ucd, ucdsamp3, testall, obj, onehex, oneprism, onepyramid,
NOTICE: onetet, subhex, TestReadMask, twohex, multispec, misc, sami, newsami,
NOTICE: specmix, spec, cpz1plt, group_test, listtypes, alltypes, wave,
NOTICE: multi_file, polyzl, csg, rocket, mmadjacency, largefile, version,
NOTICE: namescheme, efcentering, mk_nasf_pdb, ioperf, arbpoly2d, readstuff,
NOTICE: mat3d_3across, merge_block, test_mat_compression, bcastopen,
NOTICE: memfile_simple, empty, majorder, realloc_obj_and_opts, compression,
NOTICE: grab, mk_nasf_h5, testhdf5, arrayf77, arrayf90, curvef77, matf77,
NOTICE: pointf77, quadf77, ucdf77, testallf77, csgmesh, qmeshmat2df77, but
NOTICE: "testall" is a good one to start with.

make  rocket_silo.la ioperf_silo.la ioperf_pdb.la ioperf_sec2.la ioperf_stdio.la ioperf_hdf5.la arbpoly arbpoly2d array extface multi_test partial_io point group_test listtypes simple ucd ucdsamp3 testall alltypes obj onehex oneprism onepyramid onetet TestReadMask subhex twohex multispec specmix misc sami newsami spec cpz1plt merge_block mat3d_3across ucd1d wave multi_file polyzl csg rocket mmadjacency largefile dir quad version namescheme efcentering mk_nasf_pdb ioperf readstuff testfs empty majorder realloc_obj_and_opts test_mat_compression bcastopen memfile_simple   compression grab mk_nasf_h5 testhdf5 arrayf77 arrayf90 curvef77 matf77 pointf77 quadf77 ucdf77 testallf77 csgmesh qmeshmat2df77 \
  checksums testhzip testxvers test_error.py test_read.py test_write.py onehex.py testonehex testsilock testdtypes xvers_data.tar.bz2 pion0244.silo zeros.dat
make[2]: Entering directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tests'
/bin/sh ../libtool  --tag=CXX   --mode=compile mpicxx -DHAVE_CONFIG_H -I. -I..  -I./../src/silo -I./../src/silo -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include   -fPIC -MT rocket_silo.lo -MD -MP -MF .deps/rocket_silo.Tpo -c -o rocket_silo.lo rocket_silo.cxx
libtool: compile:  mpicxx -DHAVE_CONFIG_H -I. -I.. -I./../src/silo -I./../src/silo -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -fPIC -MT rocket_silo.lo -MD -MP -MF .deps/rocket_silo.Tpo -c rocket_silo.cxx  -fPIC -DPIC -o .libs/rocket_silo.o
rocket_silo.cxx: In function 'int AddRegions(DBfile*, DBmrgtree*, const char*, const char* const*, int, bool, int)':
rocket_silo.cxx:274:1: warning: no return statement in function returning non-void [-Wreturn-type]
  274 | }
      | ^
rocket_silo.cxx: In function 'int WriteSiloSingleMesh(DBfile*, DBoptlist*, const std::vector<std::__cxx11::basic_string<char> >&)':
rocket_silo.cxx:284:21: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  284 |     coordnames[0] = "X";
      |                     ^~~
rocket_silo.cxx:285:21: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  285 |     coordnames[1] = "Y";
      |                     ^~~
rocket_silo.cxx:286:21: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  286 |     coordnames[2] = "Z";
      |                     ^~~
rocket_silo.cxx:337:1: warning: no return statement in function returning non-void [-Wreturn-type]
  337 | }
      | ^
rocket_silo.cxx: In function 'int WriteSiloMultiMesh(DBfile*, DBoptlist*, const std::vector<std::__cxx11::basic_string<char> >&)':
rocket_silo.cxx:344:21: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  344 |     coordnames[0] = "X";
      |                     ^~~
rocket_silo.cxx:345:21: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  345 |     coordnames[1] = "Y";
      |                     ^~~
rocket_silo.cxx:346:21: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  346 |     coordnames[2] = "Z";
      |                     ^~~
rocket_silo.cxx: In function 'int WriteFormatReal(int, const char* const*)':
rocket_silo.cxx:491:25: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  491 |     char *dclasses[] = {"Domain0", "Domain1", "Domain2", "Domain3", "Domain4"};
      |                         ^~~~~~~~~
rocket_silo.cxx:491:36: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  491 |     char *dclasses[] = {"Domain0", "Domain1", "Domain2", "Domain3", "Domain4"};
      |                                    ^~~~~~~~~
rocket_silo.cxx:491:47: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  491 |     char *dclasses[] = {"Domain0", "Domain1", "Domain2", "Domain3", "Domain4"};
      |                                               ^~~~~~~~~
rocket_silo.cxx:491:58: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  491 |     char *dclasses[] = {"Domain0", "Domain1", "Domain2", "Domain3", "Domain4"};
      |                                                          ^~~~~~~~~
rocket_silo.cxx:491:69: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  491 |     char *dclasses[] = {"Domain0", "Domain1", "Domain2", "Domain3", "Domain4"};
      |                                                                     ^~~~~~~~~
rocket_silo.cxx: In function 'int WriteSiloMultiMesh(DBfile*, DBoptlist*, const std::vector<std::__cxx11::basic_string<char> >&)':
rocket_silo.cxx:479:1: warning: control reaches end of non-void function [-Wreturn-type]
  479 | }
      | ^
libtool: compile:  mpicxx -DHAVE_CONFIG_H -I. -I.. -I./../src/silo -I./../src/silo -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -fPIC -MT rocket_silo.lo -MD -MP -MF .deps/rocket_silo.Tpo -c rocket_silo.cxx -o rocket_silo.o >/dev/null 2>&1
mv -f .deps/rocket_silo.Tpo .deps/rocket_silo.Plo
/bin/sh ../libtool  --tag=CXX   --mode=link mpicxx  -fPIC -module -avoid-version -rpath /nowhere -L/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib  -o rocket_silo.la  rocket_silo.lo  -lhdf5  -lm
libtool: link: mpicxx  -fPIC -DPIC -shared -nostdlib /lib/../lib64/crti.o /jet/packages/spack/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-tfzxq7udz2a53dmujvasy4uz33t27iwv/lib/gcc/x86_64-pc-linux-gnu/10.2.0/crtbeginS.o  .libs/rocket_silo.o   -Wl,-rpath -Wl,/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib -Wl,-rpath -Wl,/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib -L/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib /jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib/libhdf5.so -lz -ldl -L/jet/packages/spack/opt/spack/linux-centos8-zen2/gcc-10.2.0/hwloc-1.11.11-5kevzekz7ymm3oec6thnzivbxjubreeh/lib -L/jet/packages/spack/opt/spack/linux-centos8-zen2/gcc-10.2.0/zlib-1.2.11-uw42aiy6225mpwyeha5txnsde4pxaey2/lib -L/usr/lib64 -L/jet/packages/spack/opt/spack/linux-centos8-zen2/gcc-10.2.0/openmpi-3.1.6-n7eirdskc4bywwtjdcrucflmmhxlox7j/lib -L/jet/packages/spack/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-tfzxq7udz2a53dmujvasy4uz33t27iwv/lib/gcc/x86_64-pc-linux-gnu/10.2.0 -L/jet/packages/spack/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-tfzxq7udz2a53dmujvasy4uz33t27iwv/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/jet/packages/spack/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-tfzxq7udz2a53dmujvasy4uz33t27iwv/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../.. -lmpi -lstdc++ -lm -lpthread -lc -lgcc_s /jet/packages/spack/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-tfzxq7udz2a53dmujvasy4uz33t27iwv/lib/gcc/x86_64-pc-linux-gnu/10.2.0/crtendS.o /lib/../lib64/crtn.o    -Wl,-soname -Wl,rocket_silo.so -o .libs/rocket_silo.so
libtool: link: ar cru .libs/rocket_silo.a  rocket_silo.o
libtool: link: ranlib .libs/rocket_silo.a
libtool: link: ( cd ".libs" && rm -f "rocket_silo.la" && ln -s "../rocket_silo.la" "rocket_silo.la" )
/bin/sh ../libtool  --tag=CC   --mode=compile mpicc -DHAVE_CONFIG_H -I. -I..  -I./../src/silo -I./../src/silo -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include   -fPIC -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wdeclaration-after-statement -MT ioperf_silo.lo -MD -MP -MF .deps/ioperf_silo.Tpo -c -o ioperf_silo.lo ioperf_silo.c
libtool: compile:  mpicc -DHAVE_CONFIG_H -I. -I.. -I./../src/silo -I./../src/silo -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -fPIC -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wdeclaration-after-statement -MT ioperf_silo.lo -MD -MP -MF .deps/ioperf_silo.Tpo -c ioperf_silo.c  -fPIC -DPIC -o .libs/ioperf_silo.o
libtool: compile:  mpicc -DHAVE_CONFIG_H -I. -I.. -I./../src/silo -I./../src/silo -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -fPIC -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wdeclaration-after-statement -MT ioperf_silo.lo -MD -MP -MF .deps/ioperf_silo.Tpo -c ioperf_silo.c -o ioperf_silo.o >/dev/null 2>&1
mv -f .deps/ioperf_silo.Tpo .deps/ioperf_silo.Plo
/bin/sh ../libtool  --tag=CC   --mode=link mpicc  -fPIC -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wdeclaration-after-statement -module -avoid-version -rpath /nowhere -L/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib  -o ioperf_silo.la  ioperf_silo.lo  -lhdf5  -lm
libtool: link: mpicc -shared  -fPIC -DPIC  .libs/ioperf_silo.o   -Wl,-rpath -Wl,/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib -Wl,-rpath -Wl,/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib -L/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib /jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/lib/libhdf5.so -lz -ldl -lm    -Wl,-soname -Wl,ioperf_silo.so -o .libs/ioperf_silo.so
libtool: link: ar cru .libs/ioperf_silo.a  ioperf_silo.o
libtool: link: ranlib .libs/ioperf_silo.a
libtool: link: ( cd ".libs" && rm -f "ioperf_silo.la" && ln -s "../ioperf_silo.la" "ioperf_silo.la" )
/bin/sh ../libtool  --tag=CC   --mode=compile mpicc -DHAVE_CONFIG_H -I. -I..  -I./../src/silo -I./../src/silo -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -I../src/pdb -I../src/score -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include   -fPIC -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wdeclaration-after-statement -MT ioperf_pdb_la-ioperf_pdb.lo -MD -MP -MF .deps/ioperf_pdb_la-ioperf_pdb.Tpo -c -o ioperf_pdb_la-ioperf_pdb.lo `test -f 'ioperf_pdb.c' || echo './'`ioperf_pdb.c
libtool: compile:  mpicc -DHAVE_CONFIG_H -I. -I.. -I./../src/silo -I./../src/silo -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -I../src/pdb -I../src/score -I/jet/home/bryngel/bryngel/new/MFC-develop/dependencies/build/include -fPIC -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wdeclaration-after-statement -MT ioperf_pdb_la-ioperf_pdb.lo -MD -MP -MF .deps/ioperf_pdb_la-ioperf_pdb.Tpo -c ioperf_pdb.c  -fPIC -DPIC -o .libs/ioperf_pdb_la-ioperf_pdb.o
ioperf_pdb.c:55:10: fatal error: lite_pdb.h: No such file or directory
   55 | #include <lite_pdb.h> /* below we are using lite_ functions */
      |          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:1833: ioperf_pdb_la-ioperf_pdb.lo] Error 1
make[2]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tests'
make[1]: *** [Makefile:2114: check-am] Error 2
make[1]: Leaving directory '/ocean/projects/cts160013p/bryngel/new/MFC-develop/dependencies/src/SILO/tests'
make: *** [Makefile:545: check-recursive] Error 1
```

Building Silo doesn't produce an error but it's `./configure` script does output errors interestingly enough. The following errors are in reverse-order and occurred on Bridges2:
```
configure:26813: mpicc -E -I/jet/home/leberre/MFC-develop/dependencies/build/include  conftest.c
conftest.c:60:10: fatal error: history.h: No such file or directory
   60 | #include <history.h>
```
```
configure:26754: checking readline.h presence
configure:26754: mpicc -E -I/jet/home/leberre/MFC-develop/dependencies/build/include  conftest.c
conftest.c:58:10: fatal error: readline.h: No such file or directory
   58 | #include <readline.h>
      |          ^~~~~~~~~~~~
compilation terminated.
```
```
configure:24828: checking for fpclass
configure:24828: mpicc -o conftest -fPIC -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wdeclaration-after-statement   conftest.c  >&5
/tmp/cc8kI20x.o: In function `main':
conftest.c:(.text+0xa): undefined reference to `fpclass'
collect2: error: ld returned 1 exit status
```
```
configure:24596: checking ieeefp.h presence
configure:24596: mpicc -E  conftest.c
conftest.c:37:10: fatal error: ieeefp.h: No such file or directory
   37 | #include <ieeefp.h>
      |          ^~~~~~~~~~
compilation terminated.
```
```
conftest.c:70:10: fatal error: ieeefp.h: No such file or directory
   70 | #include <ieeefp.h>
      |          ^~~~~~~~~~
compilation terminated.
```
```
configure:14565: mpicxx -E  conftest.cpp
conftest.cpp:25:10: fatal error: ac_nonexistent.h: No such file or directory
   25 | #include <ac_nonexistent.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
configure:14565: $? = 1
```
```
configure:14506: mpicxx -E  conftest.cpp
conftest.cpp:25:10: fatal error: ac_nonexistent.h: No such file or directory
   25 | #include <ac_nonexistent.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
configure:14506: $? = 1
```

--- a/dependencies/_install.sh
+++ b/dependencies/_install.sh
@@ -1,5 +1,11 @@
 #!/bin/bash
 
+# Color ANSI Escape Sequences
+FG_RED='\033[0;31m'
+FG_GREEN='\033[0;32m'
+FG_ORANGE='\033[0;33m'
+FG_NONE='\033[0m'
+
 # Exit with an error code if any command herein fails
 set -e
 set -o pipefail
@@ -65,6 +71,83 @@ cd $src_dir
         fi
     done
 cd ..
+
+# 2) Add "build/<lib, bin, include, ...>" to the system's search path
+#
+# We append the export commands to every dotfile we can find because it's
+# tricky to know which ones will be executed and when, especially if the
+# user has multiple shells installed at the same time. There are also
+# some complexities with login and non-login shell sessions.
+#
+# The code we add to the dotfiles has include guards to prevent
+# errors and redundancy. Also, we only append to the dotfiles if
+# the header guard isn't set to prevent duplicates.
+
+echo 
+echo -------------------------------------------------------
+echo --------------- Exporting Library Paths ---------------
+echo -------------------------------------------------------
+echo 
+
+found_dotfile_count=0
+found_dotfile_list_string=""
+if [[ -z "${MFC_ENV_SH_HEADER_GUARD}" ]]; then 
+    export_cmds_0="export MFC_ENV_SH_HEADER_GUARD=\"SET\""
+    export_cmds_1="export LD_LIBRARY_PATH=\"\$LD_LIBRARY_PATH:$build_dir/lib\""
+    full_dotfile_string="\n# --- [Added by MFC | Start Section] --- #\nif [[ -z \"\${MFC_ENV_SH_HEADER_GUARD}\" ]]; then \n\t$export_cmds_0 \n\t$export_cmds_1 \nfi \n# --- [Added by MFC | End Section]   --- #\n"
+
+    declare -a dotfile_paths
+
+    dotfile_names[0]=".bashrc"
+    dotfile_names[1]=".bash_profile"
+    dotfile_names[2]=".bash_login"
+    dotfile_names[3]=".profile"
+    dotfile_names[4]=".zshrc"
+    dotfile_names[5]=".cshrc"
+
+    i=1
+
+    for dotfile_name in "${dotfile_names[@]}"; do
+        dotfile_path=$HOME"/"$dotfile_name
+
+        echo -n "+--+> ($i/${#dotfile_names[@]}) $dotfile_path: "
+
+        if [[ -a "$dotfile_path" ]]; then
+            echo "Present."
+            echo -e $full_dotfile_string >> "$dotfile_path"
+
+            found_dotfile_count=$((found_dotfile_count+1))
+            
+            if [ $i -ne "1" ]; then
+                if [ $i -ne "${#dotfile_names[@]}" ]; then
+                    found_dotfile_list_string="$found_dotfile_list_string, "
+                else
+                    found_dotfile_list_string="$found_dotfile_list_string, and "
+                fi
+            fi
+
+            found_dotfile_list_string="$found_dotfile_list_string$dotfile_path"
+        else
+            echo "Absent."
+        fi
+
+        i=$((i+1))
+    done
+
+    if [ "$found_dotfile_count" -eq "0" ]; then
+        echo -e "$FG_RED\n"
+        echo "=================================================================================================="
+        echo "| [ERROR] Could not find any dotfiles where we could export the path to the installed libraries. |"
+        echo "=================================================================================================="
+        echo -e "$FG_NONE"
+        exit 1
+    fi
+
+    eval "$export_cmds_0"
+    eval "$export_cmds_1"
+else
+    echo "+--+> The MFC header guard already present, no library path will be exported."
+fi
 
 # 3) Build
 
@@ -133,7 +216,7 @@ cd $src_dir"/SILO"
     export PYTHON=python3
     export PYTHON_CPPFLAGS="$PYTHON_CPPFLAGS $(python3-config --cflags)"
 
-    # We use the following flags
+    # We use the following parameters
     # CC=mpicc CXX=mpicxx
     # So that "--with-hdf5" works...
     echo "|  |--> Configuring (./configure)..."
@@ -149,8 +232,12 @@ cd $src_dir"/SILO"
     echo "|  |--> Installing (w/ Make)..."
     make install prefix=$build_dir >> $log_filepath 2>&1
 
-echo 
+echo -e "\n$FG_GREEN"
 echo "|-----------------------------------------------------|"
 echo "|-------------- Completed Successfully ---------------|"
 echo "|-----------------------------------------------------|"
-echo 
+echo -e "\n$FG_NONE"
+
+if [ "$found_dotfile_count" -ne "0" ]; then
+    echo -e "$FG_ORANGE\n\n[WARNING] MFC's dependency install script added code to $found_dotfile_count dotfiles ($found_dotfile_list_string) in order to correctly configure your environement variables (such as LD_LIBRARY_PATH). Please restart your shell before running MFC. \n\n$FG_NONE"
+fi

--- a/dependencies/_install.sh
+++ b/dependencies/_install.sh
@@ -239,5 +239,5 @@ echo "|-----------------------------------------------------|"
 echo -e "\n$FG_NONE"
 
 if [ "$found_dotfile_count" -ne "0" ]; then
-    echo -e "$FG_ORANGE\n\n[WARNING] MFC's dependency install script added code to $found_dotfile_count dotfiles ($found_dotfile_list_string) in order to correctly configure your environement variables (such as LD_LIBRARY_PATH). Please restart your shell before running MFC. \n\n$FG_NONE"
+    echo -e "$FG_ORANGE\n\n[WARNING] MFC's dependency install script added code to $found_dotfile_count dotfiles ($found_dotfile_list_string) in order to correctly configure your environement variables (such as LD_LIBRARY_PATH). Please start a new shell session (e.g exec \$SHELL) before running MFC or source one of the listed dotfiles (e.g source ~/.bashrc). \n\n$FG_NONE"
 fi

--- a/dependencies/install.sh
+++ b/dependencies/install.sh
@@ -1,24 +1,23 @@
 #!/bin/bash
 
+# Color ANSI Escape Sequences
+FG_RED='\033[0;31m'
+FG_NONE='\033[0m'
+
 ./_install.sh "$@"
 
 exit_code=$?
 
 if [ $exit_code -ne 0 ]; then
-    echo 
-    echo
-    echo 
-    echo 
-    echo "========================================================================"
-    echo "| Fatal Error"
-    echo "| ---> Exit Code ($exit_code)"
-    echo "| ---> Some stages output logs to log/"
-    echo "| ---> Please ensure your build environment is adequate (view README.md)"
-    echo "========================================================================"
-    echo 
-    echo
-    echo 
-    echo
+    echo -e "$FG_RED\n\n\n"
+    echo "==========================================================================="
+    echo "| Fatal Error:                                                            |"
+    echo "| ---> Exit Code ($exit_code).                                             "
+    echo "| ---> Take a look at the output above.                                   |"
+    echo "| ---> Some stages output logs to log/.                                   |"
+    echo "| ---> Please ensure your build environment is adequate (view README.md). |"
+    echo "==========================================================================="
+    echo -e "\n\n\n$FG_NONE"
 fi
 
 exit $exit_code

